### PR TITLE
Service Bus and Event Hubs:  Added Templates for VirtualNetwork Rules and IPFilter Rules

### PR DESCRIPTION
**Pull request by v-Ajnava**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Added Templates for Service Bus and Event Hubs  - VirtualNetwork Rules and IPFilter Rules  : 
* 301-eventhub-namespace-ipfilter
* 301-eventhub-namespace-vnet
* 301-servicebus-namespace-ipfilter
* 301-servicebus-namespace-vnet

### CODE REVIEW

File: 301-eventhub-namespace-ipfilter/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# 301-eventhub-namespace-ipfilter
```
Comment (bmoore-msft) [inline 301-eventhub-namespace-ipfilter/README.md]: use a user friendly title

File: 301-eventhub-namespace-ipfilter/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F301-eventhub-namespace-ipfilter%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F301-eventhub-namespace-ipfilter%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+For information about using this template, see [Create a Event Hubs IP Filter rule for a Namespace]()
```
Comment (bmoore-msft) [inline 301-eventhub-namespace-ipfilter/README.md]: this link doesn't go anywhere

File: 301-eventhub-namespace-ipfilter/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
+{
+	"$schema": "http://schema.management.azure.com/schemas/2015-01-01-preview/deploymentTemplate.json#",
+	"contentVersion": "1.0.0.0",
+	"parameters": {
```
Comment (bmoore-msft) [inline 301-eventhub-namespace-ipfilter/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: 301-eventhub-namespace-ipfilter/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+				"description": "IP Mask"
+			}
+		}
+	},
+	"variables": { },
+	"resources": [{
+			"apiVersion": "2018-01-01-preview",
+			"name": "[parameters('eventhubNamespaceName')]",
+			"type": "Microsoft.EventHub/namespaces",
+			"location": "East US2",
```
Comment (bmoore-msft) [inline 301-eventhub-namespace-ipfilter/azuredeploy.json]: must be parameters('location')

File: 301-eventhub-namespace-ipfilter/azuredeploy.parameters.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "eventhubNamespaceName": {
+            "value": "GEN-UNIQUE"
+        },
+        "ipFilterRuleName": {
+            "value": "GEN-UNIQUE"
+        },
+        "ipFilterAction": {
```
Comment (bmoore-msft) [inline 301-eventhub-namespace-ipfilter/azuredeploy.parameters.json]: remove this and use a defaultValue in the template

File: 301-eventhub-namespace-ipfilter/azuredeploy.parameters.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+            "value": "Reject"
+        },
+		"IpMask": {
```
Comment (bmoore-msft) [inline 301-eventhub-namespace-ipfilter/azuredeploy.parameters.json]: remove this and use a defaultValue in the template

File: 301-eventhub-namespace-vnet/README.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# 301-eventhub-namespace-vnet
```
Comment (bmoore-msft) [inline 301-eventhub-namespace-vnet/README.md]: use a friendly title

File: 301-eventhub-namespace-vnet/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,50 @@
+{
+	"$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+	"contentVersion": "1.0.0.0",
+	"parameters": {
```
Comment (bmoore-msft) [inline 301-eventhub-namespace-vnet/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: 301-eventhub-namespace-vnet/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+		}
+	},
+	"variables": {
+		"namespaceVirtualNetworkRuleName": "[concat(parameters('eventhubNamespaceName'), concat('/', parameters('vnetRuleName')))]"
+	},
+	"resources": [{
+			"apiVersion": "2018-01-01-preview",
+			"name": "[parameters('eventhubNamespaceName')]",
+			"type": "Microsoft.EventHub/namespaces",
+			"location": "South Central US",
```
Comment (bmoore-msft) [inline 301-eventhub-namespace-vnet/azuredeploy.json]: must be parameters('location')

File: 301-eventhub-namespace-vnet/azuredeploy.parameters.json [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "eventhubNamespaceName": {
+            "value": "GEN-UNIQUE"
+        },
+        "vnetRuleName": {
+            "value": "GEN-UNIQUE"
+        },
+        "virtualNetworkSubnetId": {
+            "value": "/subscriptions/e8baea74-64ce-459b-bee3-5aa4c47b3ae3/resourceGroups/sbehvnettest/providers/Microsoft.Network/virtualNetworks/sbehvnettest/subnets/default"
```
Comment (bmoore-msft) [inline 301-eventhub-namespace-vnet/azuredeploy.parameters.json]: use GEN value (see contribution guide)

File: 301-servicebus-namespace-ipfilter/README.md [hunk 11]
```diff
@@ -0,0 +1,11 @@
+# 301-servicebus-namespace-ipfilter
```
Comment (bmoore-msft) [inline 301-servicebus-namespace-ipfilter/README.md]: looks like the same issue will be in all the files, so update the rest accordingly...

File: 301-eventhub-namespace-ipfilter/README.md (hunk 1 above)
Comment (v-Ajnava) [reply 301-eventhub-namespace-ipfilter/README.md]: resolved

File: 301-eventhub-namespace-ipfilter/README.md (hunk 2 above)
Comment (v-Ajnava) [reply 301-eventhub-namespace-ipfilter/README.md]: resolved

File: 301-eventhub-namespace-ipfilter/azuredeploy.json (hunk 3 above)
Comment (v-Ajnava) [reply 301-eventhub-namespace-ipfilter/azuredeploy.json]: resolved

File: 301-eventhub-namespace-ipfilter/azuredeploy.json (hunk 4 above)
Comment (v-Ajnava) [reply 301-eventhub-namespace-ipfilter/azuredeploy.json]: resolved

File: 301-eventhub-namespace-ipfilter/azuredeploy.parameters.json (hunk 5 above)
Comment (v-Ajnava) [reply 301-eventhub-namespace-ipfilter/azuredeploy.parameters.json]: resolved

File: 301-eventhub-namespace-ipfilter/azuredeploy.parameters.json (hunk 6 above)
Comment (v-Ajnava) [reply 301-eventhub-namespace-ipfilter/azuredeploy.parameters.json]: resolved

File: 301-eventhub-namespace-vnet/README.md (hunk 7 above)
Comment (v-Ajnava) [reply 301-eventhub-namespace-vnet/README.md]: resolved

File: 301-eventhub-namespace-vnet/azuredeploy.json (hunk 8 above)
Comment (v-Ajnava) [reply 301-eventhub-namespace-vnet/azuredeploy.json]: resolved

File: 301-eventhub-namespace-vnet/azuredeploy.json (hunk 9 above)
Comment (v-Ajnava) [reply 301-eventhub-namespace-vnet/azuredeploy.json]: resolved

File: 301-eventhub-namespace-vnet/azuredeploy.parameters.json (hunk 10 above)
Comment (v-Ajnava) [reply 301-eventhub-namespace-vnet/azuredeploy.parameters.json]: @bmoore-msft , I have added the parameters for VNET as per the contribution guide lines. The service accepts subnet ARM Id, have implemented the resourceId to fetch subnet Id, but throws errors as subnet name is undefined. can you please help on this?

File: 301-servicebus-namespace-ipfilter/README.md (hunk 11 above)
Comment (v-Ajnava) [reply 301-servicebus-namespace-ipfilter/README.md]: resolved in all the files.

**Comment by v-Ajnava** [comment]
@bmoore-msft , have resolved all the review comments, for implementation of the VNET recommended GEN-SUBNET1-NAME I am getting error. I had tried with GEN-SUBNET-NAME and GEN-SUBNET1-NAME,  both of these are throwing undefined name subnet not found error. Can you please help on this?

File: 301-eventhub-namespace-ipfilter/azuredeploy.json [hunk 12]
```diff
@@ -27,14 +28,22 @@
 			"metadata": {
 				"description": "IP Mask"
 			}
+		},
+		"Location": {
+			"type": "string",
+			"defaultValue": "East US2",
```
Comment (bmoore-msft) [inline 301-eventhub-namespace-ipfilter/azuredeploy.json]: use resourceGroup().location here and then override it in the parameters file...  Can also use "allowedValues" here if the list is small.

File: 301-eventhub-namespace-vnet/azuredeploy.json [hunk 13]
```diff
@@ -8,16 +8,29 @@
...
-		"virtualNetworkSubnetId": {
+		"subnetName": {
+			"type": "string",
+			"metadata": {
+				"description": "Name of the Virtual Network Sub Net"
+			}
+		},
+		"Location":{
 			"type": "string",
+			"defaultValue": "South Central US",
```
Comment (bmoore-msft) [inline 301-eventhub-namespace-vnet/azuredeploy.json]: use resourceGroup().location for the defaultValue - then override via parameter file

Comment (v-Ajnava) [reply 301-eventhub-namespace-vnet/azuredeploy.json]: resolved

File: 301-eventhub-namespace-ipfilter/azuredeploy.json (hunk 12 above)
Comment (v-Ajnava) [reply 301-eventhub-namespace-ipfilter/azuredeploy.json]: resolved

**Comment by v-Ajnava** [comment]
@bmoore-msft, Resolved the review comments, can you please review?

**Comment by v-Ajnava** [comment]
@bmoore-msft can you please review the changes ?

File: 301-eventhub-namespace-vnet/azuredeploy.json [hunk 14]
```diff
@@ -1,63 +1,90 @@
...
+      "sku": {
+        "name": "Standard",
+        "tier": "Standard"
+      },
+      "properties": { }
+    },
+    {
+      "apiVersion": "2017-09-01",
+      "name": "[parameters('vnetRuleName')]",
+      "location": "[resourceGroup().location]",
```
[the remark this answers is not in the document]
Comment (v-Ajnava) [reply 301-eventhub-namespace-vnet/azuredeploy.json]: resolved

File: 301-eventhub-namespace-vnet/azuredeploy.parameters.json [hunk 15]
```diff
@@ -1,18 +1,19 @@
...
-			"value": "GEN-SUBNET-NAME"
-		}
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "eventhubNamespaceName": {
+      "value": "GEN-UNIQUE-22"
+    },
+    "vnetRuleName": {
+      "value": "GEN-VNET-NAME1"
```
[the remark this answers is not in the document]
Comment (v-Ajnava) [reply 301-eventhub-namespace-vnet/azuredeploy.parameters.json]: resolved

File: 301-servicebus-namespace-vnet/azuredeploy.json [hunk 16]
```diff
@@ -1,70 +1,96 @@
...
+      "sku": {
+        "name": "Premium",
+        "tier": "Premium"
+      },
+      "properties": { }
+    },
+    {
+      "apiVersion": "2017-09-01",
+      "name": "[parameters('vnetRuleName')]",
+      "location": "[resourceGroup().location]",
```
[the remark this answers is not in the document]
Comment (v-Ajnava) [reply 301-servicebus-namespace-vnet/azuredeploy.json]: resolved

**Pull request merged**